### PR TITLE
RFC: Deprecate std::fmt::format in favor of String::format

### PR DESCRIPTION
**Pull request by kmcallister**
[Rendered view](https://github.com/kmcallister/rfcs/blob/String-format/text/0000-String-format.md)

### CODE REVIEW

**Comment by seanmonstar** [comment]
See also https://github.com/rust-lang/rfcs/pull/583#discussion_r23128698 and https://github.com/seanmonstar/rust/commit/031c5121ec2746e28e29b118c43b8b249f6976cf

File: text/0000-String-format.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+# Detailed design
+
+What it says on the tin.
+
+There's no visible change for users of the `format!` macro, except that it
+works without libstd (see [#21912](https://github.com/rust-lang/rust/pull/21912)).
+
+# Drawbacks
+
+None! Perfect RFC!
```
Comment (nagisa) [inline text/0000-String-format.md]: +1

**Comment by aturon** [comment]
ping @alexcrichton

**Comment by kmcallister** [comment]
Updated for the changes in rust-lang/rust#21912.

**Comment by alexcrichton** [comment]
@aturon and I had a [chat on IRC](https://botbot.me/mozilla/rust-internals/2015-03-05/?msg=33454322&page=12) about whether we really need a `#[stable]` API here at all. The conclusion was basically that we for now should take the most conservative route possible by stabilizing as few APIs as possible.

With [the advent of `#[allow_internal_unstable]`](https://github.com/rust-lang/rust/pull/22899) there's no actual need to have a stable API corresponding to the implementation detail of the `format!` macro. As a result, we will likely want to destabilize `fmt::format` in which case a change such as this would no longer require an RFC (but could still be done regardless).

In light of that information I'm going to close this RFC, and I think you're free to make this change as soon as #22899 lands basically.

Thanks again for the RFC, and sorry for being a little slow on the sheperding!

**Pull request closed without merging**